### PR TITLE
Feature/wipe old tokens

### PR DESCRIPTION
**Pull request by denis-yuen**
**Description**
Wipe out old github tokens
Track new TOS effort https://ucsc-cgl.atlassian.net/browse/SEAB-3765 which I think forces logout-login anyway and added note to deploy checklist 1.13 to ensure new TOS is in

**Issue**
Definitely see discussion in https://ucsc-cgl.atlassian.net/browse/SEAB-4527
Question
* worth doing this in a hotfix or wait for 1.13?
* is the user experience doing this good enough or should we force a logout?

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future

### CODE REVIEW

**Review by coverbeck — COMMENTED** [review]
Did you try resetting the tokens? The link in the ticket/email from GitHub is slightly off, [this one ](https://docs.github.com/en/rest/apps#reset-a-token/)is better).

If that works, it seems like it would be the better approach.

**Comment by denis-yuen** [comment]
(PR not quite right,  I guess our tests use the old tokens. But I think the discussion is useful)

**Comment by denis-yuen** [comment]
> Did you try resetting the tokens? The link in the ticket/email from GitHub is slightly off, [this one ](https://docs.github.com/en/rest/apps#reset-a-token/)is better).
> 
> If that works, it seems like it would be the better approach.

Haven't tried. Kinda wonder if that would be worth the trouble really. 

I'm under the impression that users would need to be forced to logout/login since we have a new TOS incoming and I think that would refresh the token if we paired it with a token delete in the same release anyway. 
https://wiki.oicr.on.ca/pages/viewpage.action?pageId=154540083

File: pom.xml [hunk 1]
```diff
@@ -451,13 +451,6 @@
                         <tag>${revision}${changelist}</tag>
                     </configuration>
                 </plugin>
-                <plugin>
```
Comment (coverbeck) [inline pom.xml:454]: Is this intentional? Just making sure, it seems unrelated to the PR.

**Review by coverbeck — APPROVED** [review]
Some random thoughts, including some real edge cases.

* If a user has linked both Google and GitHub accounts, logging in with Google won't get a new GitHub token.
* Certain parts of the CLI could stop working. I think we have a ticket for the CLI and TOS, but it's not implemented.
* This ties us to the new TOS being ready for 1.13, which presumably isn't an issue, but just noting.
* Hopefully we have good error messages in the UI and CLI if the GitHub token isn't present.

File: pom.xml (hunk 1 above)
Comment (denis-yuen) [reply pom.xml:454]: it's unrelated, looks like a duplicate entry with the one above

**Comment by denis-yuen** [comment]
> This ties us to the new TOS being ready for 1.13, which presumably isn't an issue, but just noting.

One option could be to wait till closer to the release and then consider. That way, we can hopefully avoid some redundant work.
This PR is not likely to have much of an issue with merge conflicts.

**Comment by denis-yuen** [comment]
> If a user has linked both Google and GitHub accounts, logging in with Google won't get a new GitHub token.

this is true, but I do think it's a real edge case because they would need to login via google and then do a refresh (not just work with a github app enabled workflow) and even then the message is just 


> Hopefully we have good error messages in the UI and CLI if the GitHub token isn't present

think I either got this message or a very similar one at the top
https://github.com/dockstore/dockstore-ui2/blob/2.9.2/src/app/myworkflows/my-workflow/my-workflow.component.html#L183-L187

**Review by svonworl — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]

**Comment by denis-yuen** [comment]
related, created https://ucsc-cgl.atlassian.net/browse/SEAB-4766

**Comment by denis-yuen** [comment]
Looks like the library we're using doesn't support token resets, whether updated or not. 
Subscribing to https://github.com/hub4j/github-api/issues/1373

**Comment by denis-yuen** [comment]
For tomorrow, trying to hit the reset tokens API directly. 
No dice, with a bunch of different variants, I just keep on ending up hitting 404s. 
Not sure if I'm hitting the right endpoint, submitting the correct body or what. Sigh.
